src/utils.ts

```diff
@@ -186,14 +195,17 @@ export function isJSONString(secretValue: string): boolean {
  * Transforms the secret name into a valid environmental variable name
  * It should consist of only upper case letters, digits, and underscores and cannot begin with a number
  */
-export function transformToValidEnvName(secretName: string): string {
+export function transformToValidEnvName(secretName: string, nameTransformation?: TransformationFunc): string {
     // Leading digits are invalid
     if (secretName.match(/^[0-9]/)){
         secretName = '_'.concat(secretName);
     }
 
     // Remove invalid characters
-    return secretName.replace(/[^a-zA-Z0-9_]/g, '_').toUpperCase()
+    secretName = secretName.replace(/[^a-zA-Z0-9_]/g, '_');
+
+    // Apply the name transformation. When no transformation is defined fallback to the "uppercase" transformation
+    return nameTransformation ? nameTransformation(secretName) : secretName.toUpperCase();
```

Comment:
Can we set nameTransformation to uppercase earlier in the call stack if not set? (In `index.ts` for example)

Reply:
The `parseTransformationFunction` returns non-nullable, and the real github action execution never encounters the null value. 
https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/05c8c02fa716d39f8e54243b824ce1abf69acf4d/src/utils.ts#L257

The reason for `nameTransformation` in every util function being nullable 
https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/05c8c02fa716d39f8e54243b824ce1abf69acf4d/src/utils.ts#L26

is that we have 46 existing unit tests. Each of these tests expects uppercase transformation. Having a newly introduced transformation as optional argument is a way for me to not modify the contract. If this is bothersome I can remove nullability and go *act* by *act* and append `x => x.toUpperCase()`, but I don't think that will make things nicer.